### PR TITLE
Represent border

### PR DESCRIPTION
**Pull request by jackdavidweber**

### CODE REVIEW

File: gym_snake/envs/snakeGameGym.py [hunk 1]
```diff
@@ -88,39 +88,56 @@ def pos_on_board(self, pos):
 		# Otherwise, pos is on board
 		return True
 
-	def get_board(self) -> np.ndarray:
+	def get_board(self, represent_border=False) -> np.ndarray:
```
Comment (jackdavidweber) [inline gym_snake/envs/snakeGameGym.py]: add type

File: gym_snake/envs/snakeGameGym.py [hunk 2]
```diff
@@ -88,39 +88,56 @@ def pos_on_board(self, pos):
...
 		2 is space with snake body in it
 		3 is space with snake head in it
 		"""
 		# Initializes empty board
 		board = np.zeros([self.rows, self.cols], dtype=int)
+		border_space = 0
+
+		if represent_border:
+			board = np.zeros([self.rows+2, self.cols+2], dtype=int)
+			border_space = 1  # FIXME: can be replaced with int(represent_border)
```
Comment (jackdavidweber) [inline gym_snake/envs/snakeGameGym.py]: replace with int(represent_border)

File: trainTestReinforcementAlgorithm.py [hunk 3]
```diff
@@ -60,6 +61,7 @@ def trainRL(
         visualize_training (bool): We don't want to visualize the training process.
         visualization_fps (int): Default to a high value for training speed if training is visualized.
         reward_function (fn(reward_dict)->int): Set reward function to be used in training. Reward functions are defined in snakeRewardFuncs.py
+        represent_border (bool):
```
Comment (jackdavidweber) [inline trainTestReinforcementAlgorithm.py]: add docstring

File: trainTestReinforcementAlgorithm.py [hunk 4]
```diff
@@ -100,7 +103,8 @@ def testRL(
     max_moves_no_fruit=0, # Set number of allowed moves without fruit consumption before ending the game. Any non-poitive number corresponds to no limit.
     visualize_testing=True, # Set to true in order to see game moves in pygame. Should be false if run on server.
     visualization_fps=30, # Set frames per second of testing visualization.
-    reward_function=RewardFuncs.basic_reward_func # Set reward function to be used in training. Reward functions are defined in snakeRewardFuncs.py
+    reward_function=RewardFuncs.basic_reward_func, # Set reward function to be used in training. Reward functions are defined in snakeRewardFuncs.py
+    represent_border=False,
```
Comment (jackdavidweber) [inline trainTestReinforcementAlgorithm.py]: Add comment explaining represent_border

**Review by jackdavidweber — COMMENTED** [review]
Some style fixes that I need to implement

File: gym_snake/envs/snakeGameGym.py (hunk 2 above)
Comment (davidd-55) [reply gym_snake/envs/snakeGameGym.py]: yup agree with this for cleanliness 👍

File: trainTestReinforcementAlgorithm.py (hunk 3 above)
Comment (davidd-55) [inline trainTestReinforcementAlgorithm.py]:
```suggestion
        represent_border (bool): boolean flag for whether or not to represent the border in observation.
```

File: trainTestReinforcementAlgorithm.py (hunk 4 above)
Comment (davidd-55) [inline trainTestReinforcementAlgorithm.py]:
```suggestion
    represent_border=False, # Set a boolean flag for whether or not to represent the border in observation.
```

**Review by davidd-55 — APPROVED** [review]

**Comment by davidd-55** [comment]
LGTM! I added in suggestions for some of the comments you already left, but it looked functional when running it locally so all good

**Pull request merged**